bugsnag-android-core/src/main/java/com/bugsnag/android/ImmutableConfig.kt

```diff
@@ -160,5 +161,38 @@ internal fun sanitiseConfiguration(
             configuration.buildUuid = buildUuid
         }
     }
+
+    // populate releaseStage
+    if (configuration.releaseStage == null) {
```

Comment:
We're only entering this block if `releaseStage` is null in config, so the check in `guessReleaseStage` is redundant.

Reply:
The check is not redundant in `guessReleaseStage` as `config.releaseStage` is a mutable property whose value could have been mutated on another thread - the Kotlin compiler will not allow us to remove the null-check from that method as-is. We could refactor the method to take a `String` only and copy `config.releaseStage` to a local variable, but I don't think there's much benefit in doing so.